src/main/java/com/minecolonies/rankup/util/RankingUtils.java

```diff
@@ -50,21 +39,21 @@ public static void timeUp(final Player player, final Rankup plugin)
 
     public static void balanceCheck(final Player player, final Rankup plugin)
     {
-        final List<String> playerGroups = CoreModule.perms.getPlayerGroupIds(player);
+        final List<String> playerGroups = plugin.perms.getPlayerGroupIds(player);
+
+        final GroupsConfig groupsConfig = plugin.configUtils.getGroupsConfig(player);
 
         //Check if player is in disabled group.
-        for (final Subject subject : CoreModule.perms.getDisabledGroups())
+        for (final String subject : plugin.perms.getDisabledGroups())
         {
-            if (playerGroups.contains(subject.getIdentifier()))
+            if (playerGroups.contains(subject))
             {
+                plugin.getLogger().info("disabled group: " + subject);
                 return; //IF the player is withing a disabled group, stop here.
             }
         }
 
-        final GroupsConfig groupsConfig = (GroupsConfig) plugin.getAllConfigs().get(GroupsConfig.class);
-
-        final String highestGroup = CoreModule.perms.getPlayerHighestRankingGroup(player);
-        final String nextGroup = CoreModule.perms.getNextGroup(groupsConfig.groups.get(highestGroup).rank);
+        final String nextGroup = plugin.perms.getNextGroup(player);
 
         if (plugin.econ != null && !nextGroup.equals("") && groupsConfig.groups.get(nextGroup).moneyNeeded != 0)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the "" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1132)